src/dockerRunner.ts

```diff
@@ -72,39 +72,58 @@ export class DockerRunner extends TerminalBaseRunner {
-                        vscode.window.showTextDocument(doc);
-                    });
-                }
-            });
-            TerminalExecutor.runInTerminal(initCmd, Constants.AnsibleTerminalName + ' ' + Option.docker, true, subCmds, 180, false, function (terminal, interval) {
-                if (terminal) {
-                    require('child_process').exec('docker ps --filter name=' + terminalId, (err, stdout, stderr) => {
-                        if (err || stderr) {
-                            console.log('err: ' + err + ' ' + stderr);
-                            return;
-                        }
-                        if (stdout) {
-                            // check if docker container is up
-                            if (stdout && stdout.indexOf('Up ') > -1) {
-
-                                // then send other commands to terminal
-                                for (let text of subCmds) {
-                                    terminal.sendText(text);
-                                }
-                                terminal.show();
-                                if (interval) {
-                                    clearInterval(interval);
-                                }
+            if (!utilities.isCredentialConfigured()) {
+                const cancelItem: vscode.MessageItem = { title: "Not Now" };
+                const promptMsg = 'Please configure cloud credentials at ' + utilities.getCredentialsFile() + ' for first time';
+                
+                vscode.window.showWarningMessage(promptMsg, msgOption, msgItem, cancelItem).then(response => {
+                    vscode.workspace.getConfiguration('ansible').update('credentialsConfigured', true);
+                    if (response === msgItem) {
+                        vscode.workspace.openTextDocument(utilities.getCredentialsFile()).then(doc => {
+                            vscode.window.showTextDocument(doc);
+                        });
+                    } else if (response === cancelItem) {
+                        this.startTerminal(terminalId, initCmd, Constants.AnsibleTerminalName + ' ' + Option.docker, true, subCmds, 180, false);
+                    }
+                });
+            } else {
+                const openItem: vscode.MessageItem = { title: "Open File" };
+                vscode.window.showInformationMessage('Use cloud credential file ' + utilities.getCredentialsFile(), msgOption, msgItem, openItem).then(response => {
+                    if (response === openItem) {
+                        vscode.workspace.openTextDocument(utilities.getCredentialsFile()).then(doc => {
+                            vscode.window.showTextDocument(doc);
+                        });
+                    }
+                });
+                this.startTerminal(terminalId, initCmd, Constants.AnsibleTerminalName + ' ' + Option.docker, true, subCmds, 180, false);
+            }
```

Comment:
is it intentional like this? from the code it seems like terminal will be started and dialog will be displayed at the same time. is there any way to suppress this dialog? like "don't show it again"?

Reply:
yes. it's intentional. 

when user run playbook in docker:

1.  for first time, a warning message prompted with message " Please configure cloud credentials file xxx". with 2 options: ok, not now. if click not now, continue running playbook. if click ok, open credential file.
2.  after first time, an information message prompted with message "Use cloud credential file xxx', with 2 options: ok, open file. click on each option, continuing running playbook. An information msg could be safely ignored, it will not block user's operation.